### ISSUE TITLE
Feature: Support translations

### ISSUE THREAD

**Issue by terichadbourne** [ProtoSchool/protoschool.github.io#268]
Parts of the IPFS project are structured in a way that accommodates localization, with volunteer translators working via Transifex. ([Learn more here](https://github.com/ipfs/community/blob/master/CONTRIBUTING.md#translations) about contributing translations.) 

Once we have ProtoSchool's structure more stable (which will depend on completing our audit of online coding education platforms and determining whether we're migrating), it would be great to revamp our structure in a way that would support translations on the ProtoSchool website as well. 

This is not a change we can make immediately, but I'm opening the issue now so folks can follow it if they're interested in knowing when translations are possible.

**Comment by terichadbourne** [comment]
For future reference, here's a [PDF of a presentation in Mandarin](https://github.com/ProtoSchool/shanghai/blob/b62173453f2eddf6e1bac5944b3292221124b2bd/Events/20190728-4thSession/%E8%BE%BE%E5%AE%87%EF%BC%9AIPFS%E5%8E%BB%E4%B8%AD%E5%BF%83%E5%8C%96%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84.pd) by the Shanghai chapter that covers the content of our [Decentralized Data Structures](https://proto.school/#/data-structures/) tutorial.  Thanks to @steven004 for sharing this presentation by Yu Da  (a senior engineer from IPFS-Force) which took place July 28, 2019.

**Comment by steven004** [comment]
I am putting the [Chinese version of Decentralized Data Structures](https://github.com/ProtoSchool/shanghai/blob/master/Events/20190728-4thSession/Data_Structures_ChineseVersion.md) Tutorial into Shanghai Chapter repo.

And, also welcome to watch the video of the training from [here](https://www.youtube.com/watch?v=mUwoolN2O5o).

**Comment by terichadbourne** [comment]
I just found another translation of the Decentralized Data Structures tutorial into Chinese [here](https://learnblockchain.cn/article/744).

**Comment by terichadbourne** [comment]
I accidentally created this as a duplicate issue but have been referring folks, here, so now copying in some notes from #39: 

from @lidel on 9/10/18: 

@mikeal we should add it to https://www.transifex.com/ipfs/public/ :)

Steps I see:

1. Create a PR which moves English strings to separate source files, preferably in JSON-ICU format (to make it framework-agnostic and integrate with Transifex without conversion step). 
   Some prior art:
    - In new WebUI we're using `i18next` with `i18next-icu`, [the setup is quite straightforward](https://github.com/ipfs-shipyard/ipfs-webui/blob/a67135c0a5041ed5c7a6d5216e9b664cb5336726/src/i18n.js).
    - There is an open PR adding i18n to IPLD Explorer that you can check as well: https://github.com/ipfs-shipyard/ipld-explorer-components/pull/1/
    - Those apps use React, but `i18next` bits will be mostly the same. I see there is also a helper for Vue: [@panter/vue-i18next](https://www.npmjs.com/package/@panter/vue-i18next), but I had no experience with it.
1. In the same PR add `.tx/config` and create a project at Transifex (ping @lidel)
1. Merge to master
1. Set up Transifex to automatically check source strings from `master` for updates 

Step 1 is a chore but If you create PR with that, I can set up the rest :)

For more info about Transifex part, see References in https://github.com/ipfs/ipfs-gui/issues/50

**Comment by bertrandfalguiere** [comment]
> @mikeal we should add it to https://www.transifex.com/ipfs/public/ :)

Yes please! There are a lot of translators and growing activity on Transifex. This will help the growth of IPFS, etc in non-English speaking countries. Chinese, Spanish, Arabic, French and German are much needed, to name a few. 

Also, there is a potential risk associated with waiting too long: external websites will start summing up proto.school with their own translations, which could slightly differ, leading up to confusion for learners. Putting it up on Transifex also ensures it is consistent with the translations found on ipfs.io, IPFS Desktop, etc.

**Comment by terichadbourne** [comment]
Totally agree, @bertrandfalguiere, and will keep folks updated here. Preparing for translation will require a major overhaul of our codebase, which we're prioritizing alongside some other large projects on the horizon.

In the meantime, we're using our user survey to capture language preferences and are building new features with translations files so that there will be less to overhaul when the time comes.

Can't wait to get the community's support with translations once we've made the necessary structural changes!
